### ISSUE TITLE
Cube3D Plugin is very slow to respond...

### ISSUE THREAD

**Issue by bigplayer-ai** [newlooper/VirtualSpace#44]
Hey,
Cube3D Plugin is very slow to respond... after the desktop is changed it shows the cube3d transition if even showed. (Sometimes it does not even get triggered).
After using cube3D multiple times, Cube3D seems to get warmed up and start working properly.

**Comment by newlooper** [comment]
This has nothing to do with Cube3D, or even with any application, it's due to Windows memory management mechanism - program that not interact with the user for a long time will swap out of physical memory to disk (`pagefile`, `swapfile`...), and when the user interacts with that program again, swap from disk to physical memory, then the program run smoothly (use your word: get warmed up).

The slowness or even unresponsiveness you experience stems from this, and the more obvious the condition is if the destination of the swap out is a HDD.

In other words, if Cube3D is not idle for a long period of time (eg. just started or interacting with users very frequently), the above phenomenon basically does not happen.

It is often debated whether to disable `pagefile` in order to avoid this, but in general it is better to keep the system defaults and let Windows take care of programs.

**Comment by bigplayer-ai** [comment]
If I may ask, Do you work full time software engineer? You seem to know a lot of computer engineering stuff.Best Regards,On 28 Feb 2023, at 19:38, Dylan Cheng ***@***.***> wrote:﻿
This has nothing to do with Cube3D, or even with any application, it's due to Windows memory management mechanism - program that not interact with the user for a long time will swap out of physical memory to disk (pagefile, swapfile...), and when the user interacts with that program again, swap from disk to physical memory, then the program run smoothly (use your word: get warmed up).
The slowness or even unresponsiveness you experience stems from this, and the more obvious the condition is if the destination of the swap out is a HDD.
In other words, if Cube3D is not idle for a long period of time (eg. just started or interacting with users very frequently), the above phenomenon basically does not happen.
It is often debated whether to disable pagefile in order to avoid this, but in general it is better to keep the system defaults and let Windows take care of programs.

**Comment by newlooper** [comment]
.net programming is my hobby, my job is a software architect, mostly using Java, PHP, C/C++, Python, Go, JScript...

**Comment by bigplayer-ai** [comment]
How can I disable pagefile in order to avoid this?

**Comment by newlooper** [comment]
its not a good idea, but it can be done.

https://www.intowindows.com/how-to-delete-pagefile-sys-in-windows-10/
https://www.howtogeek.com/126430/htg-explains-what-is-the-windows-page-file-and-should-you-disable-it/

do it when and only when you know what are you doing AND you have enough physical memory (eg. 32G、64G……)

**Comment by bigplayer-ai** [comment]
I want it only for the Cube3D plugin, Is this possible?

> It is not possible to disable pagefile for a specific program through Windows settings. [However, if you have access to the source code of the program, you can use mlockall (MCL_CURRENT | MCL_FUTURE) at the start of the program to prevent paging](https://stackoverflow.com/questions/2360116/how-to-prevent-paging-for-one-program-process)[1](https://stackoverflow.com/questions/2360116/how-to-prevent-paging-for-one-program-process). [Be aware that this will increase the chances of memory allocations made by the process failing](https://stackoverflow.com/questions/2360116/how-to-prevent-paging-for-one-program-process)[1](https://stackoverflow.com/questions/2360116/how-to-prevent-paging-for-one-program-process).

**Comment by newlooper** [comment]
no, this require program remake by another language.
